burr/core/application.py

```diff
@@ -268,12 +282,11 @@ def step(self, inputs: Optional[Dict[str, Any]] = None) -> Optional[Tuple[Action
         :param inputs: Inputs to the action -- this is if this action requires an input that is passed in from the outside world
         :return: Tuple[Function, dict, State] -- the function that was just ran, the result of running it, and the new state
         """
-
-        try:
-            out = self._step(inputs=inputs, _run_hooks=True)
-            return out
-        finally:
-            self._increment_sequence_id()
+        # we need to increment the sequence before we start computing
```

Comment:
Does this break the UI or another part of the model? The removed part was done specifically so:

1. The sequence ID was always the current step being executed. Every call at the step would be 
2. It was always incremented right before the next one
3. We didn't have to start at `-1`, so the sequence ID would start at `0`

So, it would start at 0, and every step would increment it at the *very end* so the last one would start it. I don't mind how it works, but my only requirements are that (a) it starts at zero and (b) it gets incremented regardless of failure and (c) we only increment it once per `step()/_step()` call.

Reply:
Also why would it get stuck? That doesn't follow from incrementing beforehand...

Reply:
Answer -- we start at the beginning and it all gets simpler